dbms/src/Storages/Page/V3/LogFile/LogWriter.cpp

```diff
@@ -139,7 +138,21 @@ void LogWriter::addRecord(ReadBuffer & payload, const size_t payload_size, const
             type = recycle_log_files ? Format::RecordType::RecyclableLastType : Format::RecordType::LastType;
         else
             type = recycle_log_files ? Format::RecordType::RecyclableMiddleType : Format::RecordType::MiddleType;
-        emitPhysicalRecord(type, payload, fragment_length);
+        // Check available space in write_buffer before writing
+        if (buffer_size - write_buffer.offset() < fragment_length + header_size)
+        {
+            flush(write_limiter, /* background */ false);
+        }
+        try
+        {
+            emitPhysicalRecord(type, payload, fragment_length);
+        }
+        catch (...)
+        {
+            auto message = getCurrentExceptionMessage(false);
```

Comment:
should we call `getCurrentExceptionMessage` with `with_stacktrace=true` to get more information what happen here?

Reply:
Change to `with_stacktrace=true`.